Kernel/Devices/FloppyDiskDevice.cpp

```diff
@@ -238,7 +244,7 @@ bool FloppyDiskDevice::write_sectors_with_dma(u16 lba, u16 count, const u8* inbu
     for (int i = 0; i < 3; i++) {
         // Now actually send the command to the drive. This is a big one!
         send_byte(FLOPPY_MFM | FLOPPY_MT | static_cast<u8>(FloppyCommand::WriteData));
-        send_byte(head << 2 | is_slave() ? 1 : 0);
+        send_byte(head << 2 | drivesel);
```

Comment:
This could also be written this way:
```suggestion
        send_byte(head << 2 | is_slave());
```